app/controllers/tool_drafts_controller.rb

```diff
@@ -7,13 +7,95 @@ class ToolDraftsController < BaseDraftsController
   before_action :set_form, only: [:edit, :update, :show]
   before_action :set_current_form, only: [:edit]
   before_action :set_preview, only: [:show]
+  before_action :set_resource, only: [:download_json]
+  before_action :proposal_approver_permissions, except: [:download_json]
+  skip_before_action :ensure_user_is_logged_in, only: [:download_json]
 
   def edit
     super
 
     set_tool_keywords if @current_form == 'descriptive_keywords'
   end
 
+  def download_json
+    if Rails.env.development?
+      render json: JSON.pretty_generate(get_resource.draft) if Rails.env.development?
+      return
+    end
+
+    authorization_header = request.headers['Authorization']
+
+    if authorization_header.nil?
+      render json: JSON.pretty_generate({'error': 'unauthorized'}), status: 401
+      return
+    end
+    token = authorization_header.split(' ', 2)[1] || ''
+
+    # Handle EDL authentication
+    if authorization_header.start_with?('Bearer')
+      if Rails.configuration.proposal_mode
+        token_response = cmr_client.validate_dmmt_token(token)
+      else
+        token_response = cmr_client.validate_mmt_token(token)
+      end
+      token_info = token_response.body
+      token_info = JSON.parse token_info if token_info.class == String # for some reason the mock isn't return hash but json string.
+      urs_uid = token_info['uid']
+    else
+      render json: JSON.pretty_generate(get_resource.draft)
+      return
+      # Todo: We need to handle verifying a launchpad token.
```

Comment:
@htranho 

Hello! A small nit -- should this be added as a GitHub issue rather than as an in-line todo (hopefully to make more discoverable and easier to track/organize)?

Reply:
This method is a copy of one from collection controller. Devs are aware of it.